sheep/src/format/mod.rs

```diff
@@ -1,10 +1,14 @@
+use serde::Serialize;
+
 #[cfg(feature = "amethyst")]
 pub mod amethyst;
 
+pub mod named;
+
 use SpriteAnchor;
 
 pub trait Format {
-    type Data;
+    type Data: Serialize;
```

Comment:
What's the reason for adding this constraint? I don't see why you would always want to serialize `Data`, you might also use the library to just have the information in-memory or to construct texture atlasses

Reply:
in `sheep_cli/src/main.rs` the `do_pack` functions produces the serialized output, so i thought this fit that use case. splitting out a `write_serialized` function from `do_pack` could get rid of this constraint.

Reply:
i think we need to either keep the constraint or bring in a `serde` dependency for `sheep_cli` in order to support multiple formats, since the format can't be known at compile time.

Reply:
I'd rather make the cli more complicated than constrain the library, so I'd agree with bringing in dependencies for the cli and/or splitting out the serialize function.